[pruner] add Conv2d support

kazhou opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61778

Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

kazhou committed:
[pruner] add Conv2d support

Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]

jerryzh168 commented on test/ao/sparsity/test_pruner.py:
```diff
@@ -11,7 +11,7 @@
 logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
 
-class Model(nn.Module):
+class Linear(nn.Module):
```
nit: using the same name as nn module (since we have nn.Linear) might be a bit confusing, maybe we can have a different name like LinearA

jerryzh168 commented on test/ao/sparsity/test_pruner.py:
```diff
@@ -93,6 +93,20 @@ def forward(self, x):
         return x
 
 
+class Conv2d(nn.Module):
```
same here

jerryzh168 reviewed (approved):
LG

kazhou committed:
Update on "[pruner] add Conv2d support"


Adding Conv2d as supported modules for the pruner. Previously the pruner only supported Linear layers. This addition includes:
- adding a Conv2d activation reconstruction forward hook to match Conv2d weight shapes
- in `prepare`, checking the type of the module and using the corresponding activation forward hook

Differential Revision: [D29719045](https://our.internmc.facebook.com/intern/diff/D29719045/)

[ghstack-poisoned]
